src/components/FilterButton/FilterButton.jsx

```diff
@@ -2,20 +2,24 @@ import { useState } from "react";
 
 export const FilterButton = ({filterOrigin, origin}) => {
   const [buttonColor, setButtonColor] = useState('white');
+  const [buttonSelected, setButtonSelected] = useState(false);
+
   const markButtonSelected = () => {
-    buttonColor === "white" ? setButtonColor("lightBlue") : setButtonColor("white");
+    buttonSelected ? setButtonColor("white") : setButtonColor("lightBlue");
   }
+
   return (
     <button key={origin} className="filterButton"
-    key={origin}
-    onClick={() => {
-    filterOrigin({origin}); 
-    markButtonSelected();
-  }} 
-  style={{
-    backgroundColor: buttonColor,
-    borderRadius: '8px',
-    marginBottom: '20px'
-  }}>{origin}</button>
+      onClick={() => {
+        const isButtonSelected = setButtonSelected(!buttonSelected);
```

Comment:
BTW, `setButtonSelected` will indeed invert properly since the component re-renders when `buttonSelected` is changed, but let's consider a scenario where there are multiple sideeffects and you don't yet have the latest value of `buttonSelected`, but you need it. You can actually get the `buttonSelected` value from the set hook itself:

`setButtonSelected (selected => !selected)`

Reply:
Blowing my mind here